### PR TITLE
tests: switch to `uvu/assert` and its methods

### PR DESCRIPTION
**Pull request by XhmikosR**
/CC @lukeed

### CODE REVIEW

File: test/quicklink.spec.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ mainSuite('should prefetch in-viewport links correctly (UMD)', async () => {
...
   await page.goto(`${server}/test-basic-usage.html`);
   await sleep(1000);
-  assert.equal(Array.isArray(responseURLs), true);
-  assert.equal(responseURLs.includes(`${server}/1.html`), true);
-  assert.equal(responseURLs.includes(`${server}/2.html`), true);
-  assert.equal(responseURLs.includes(`${server}/3.html`), true);
+  assert.is(Array.isArray(responseURLs), true);
+  assert.is(responseURLs.includes(`${server}/1.html`), true);
+  assert.is(responseURLs.includes(`${server}/2.html`), true);
+  assert.is(responseURLs.includes(`${server}/3.html`), true);
```
Comment (lukeed) [inline test/quicklink.spec.js:40]: low prio: These can be `assert.ok` which checks truthiness
```suggestion
  assert.ok(Array.isArray(responseURLs));
  assert.ok(responseURLs.includes(`${server}/1.html`));
  assert.ok(responseURLs.includes(`${server}/2.html`));
  assert.ok(responseURLs.includes(`${server}/3.html`));
```

File: test/quicklink.spec.js [hunk 2]
```diff
@@ -34,10 +34,10 @@ mainSuite('should prefetch in-viewport links correctly (UMD)', async () => {
   });
   await page.goto(`${server}/test-basic-usage.html`);
   await sleep(1000);
-  assert.equal(Array.isArray(responseURLs), true);
```
Comment (lukeed) [inline test/quicklink.spec.js:37]: Or can use another method :P

```suggestion
  assert.instance(responseURLs, Array);
```

File: test/quicklink.spec.js [hunk 3]
```diff
@@ -139,11 +139,11 @@ mainSuite('should only prefetch links of same origin (default)', async () => {
...
+  assert.is(Array.isArray(responseURLs), true);
   // => origins: [location.hostname] (default)
-  assert.equal(responseURLs.includes(`${server}/2.html`), true);
-  assert.equal(responseURLs.includes('https://example.com/1.html'), false);
-  assert.equal(responseURLs.includes('https://github.githubassets.com/images/spinners/octocat-spinner-32.gif'), false);
+  assert.is(responseURLs.includes(`${server}/2.html`), true);
+  assert.is(responseURLs.includes('https://example.com/1.html'), false);
+  assert.is(responseURLs.includes('https://github.githubassets.com/images/spinners/octocat-spinner-32.gif'), false);
```
Comment (lukeed) [inline test/quicklink.spec.js:146]: low prio alternatives

```suggestion
  assert.not.ok(responseURLs.includes('https://example.com/1.html'));
  assert.not.ok(responseURLs.includes('https://github.githubassets.com/images/spinners/octocat-spinner-32.gif'));
```

**Review by lukeed — APPROVED** [review]
lgtm :)

**Comment by XhmikosR** [comment]
TBH I was using the previous method out of habit for strictness, but I agree `ok`/`not.ok` is enough here.

I'll apply your suggestions, thanks @lukeed!

File: test/quicklink.spec.js (hunk 2 above)
Comment (XhmikosR) [reply test/quicklink.spec.js:37]: I didn't see this documented and being that Arrays are treated weird in JS I thought you didn't cover this, but indeed you cover it with `constructor.name`. :)

Comment (lukeed) [reply test/quicklink.spec.js:37]: It is :) just passes thru an instanceof check

**Comment by XhmikosR** [comment]
@lukeed I think I addressed all of your comments :)

**Comment by XhmikosR** [comment]
BTW do you have any ideas about the test I had to skip (happens on master too)?

https://github.com/GoogleChromeLabs/quicklink/blob/dc0e4511072efb04904c08de93c6203cc9963f65/test/quicklink.spec.js#L232

**Comment by lukeed** [comment]
Im not sure, I havent been with the project for a while & I can't get any tests to pass locally either

**Comment by XhmikosR** [comment]
Weird, I'm even mostly on Windows and tests pass here too as on CI using
Node.js 18. I suppose you've done the usual clean npm packages and stuff,
right?

**Comment by lukeed** [comment]
yup, fresh clone, node 18, npm ci, etc

**Comment by XhmikosR** [comment]
Could be something network related? Because the tests rely on external
requests, which isn't very nice 😜

Another thing could be the timeouts between each test that's not enough for
some reason on your machine.

Let me know if I can help in any way and keep up the good work!

**Review by lukeed — APPROVED** [review]

**Pull request merged**